keras/models.py

```diff
@@ -972,6 +972,13 @@ def input_validation(generator_output):
                                 '2 or 3 elements.')
             return X, y, sample_weight
 
+        if do_validation:
+            X_val, y_val, sample_weight_val = input_validation(validation_data)
+            sample_weight_val = standardize_weights(y_val, sample_weight=sample_weight_val,
+                                                    sample_weight_mode=self.sample_weight_mode)
```

Comment:
For `Graph`, I need the `sample_weight` to be a `dict` and I create a list to so it has the right format. I keep the dict so it could be passed to evaluate. Should I move the normalization of weights to `input_validation` and return a `list` and a `dict`?